metaflow/plugins/argo/argo_decorator.py

```diff
@@ -0,0 +1,131 @@
+import json
+from metaflow.decorators import FlowDecorator
+from metaflow.decorators import StepDecorator
+from .argo_exception import ArgoException
+
+
+class ArgoFlowDecorator(FlowDecorator):
+    """
+    Flow decorator for Argo Workflows, that sets a default
+    for all steps in the flow.
+    To use, add this decorator directly on top of your Flow class:
+    ```
+    @argo_base(image='python:3.8')
+    class MyFlow(FlowSpec):
+        ...
+    ```
+
+    Any step level argo decorator will override any setting by this decorator.
+
+    Parameters
+    ----------
+    image: string
+        Docker image is used for Argo Workflows template.
+        If not specified, a default image mapping to
+        a base Python/ML container is used.
+    labels: json
+        Labels to be attached to the workflow template.
+    annotations: json
+        Annotations to be attached to the workflow template.
+    env: list
+        Additional environment variables to be set in all steps.
+        Follows the k8s spec.
+    envFrom: list
+        Additional environment variables to be set in all steps.
+        Follows the k8s spec.
+    imagePullSecrets: list
+        Credentials for pulling docker images.
+    """
+    name = 'argo_base'
+    defaults = {
+        'image': None,
```

Comment:
This is a broader q beyond this PR, so just thinking aloud: there probably should be a more consistent way to specify docker images between `@batch`, `@argo` and potential future execution envs based on Docker. For example, right now for `@batch` default image is a configuration setting, not flow-level decorator parameter. There is also a docker image param introduced in #463 for AWS Lambda.

Reply:
yes, accepting docker images from the config/envar is on our table.
There is indeed a broader topic.  Recently we had a discussion internally whether we should re-use Metaflow's @batch config names or introduce @argo's alternatives, e.g. `SFN_STATE_MACHINE_PREFIX`.  We opted to have a separate `ARGO_WORKFLOW_PREFIX` for similar functionality to avoid users' confusion.  The same applies to the `BATCH_CONTAINER_IMAGE`.
Ideally, we'd have neutral names for common config variables like `METAFLOW_CONTAINER_IMAGE` which could be re-used in different Metaflow plugins.  Even by @batch at some point though @batch would need to support the old name for backward compatibility.

Reply:
Started a discussion here https://github.com/Netflix/metaflow/issues/489, please chime in